lighthouse-core/gather/connections/sw.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Connection = require('./connection.js');
+
+class SWSession extends Connection {
+  constructor(sessionId, targetInfo, parentConnection) {
+    super();
+    this._targetInfo = targetInfo;
+    this._sessionId = sessionId;
+    this._parent = parentConnection;
+
+    this._parent.on('notification', this._onEvent.bind(this));
```

Comment:
my only gripe with the overall setup is this `notification` listener bit since it feels like a hacky way to override a method, can it not be done with regular override and super call?